jwql/shared_tasks/shared_tasks.py

```diff
@@ -0,0 +1,516 @@
+    logging.info("Pipeline Call for {} requesting {}".format(input_file, ext_or_exts))
+    config = get_config()
+    send_path = os.path.join(config["transfer_dir"], "incoming")
+    ensure_dir_exists(send_path)
+    receive_path = os.path.join(config["transfer_dir"], "outgoing")
+    ensure_dir_exists(receive_path)
+    
+    input_path, input_name = os.path.split(input_file)
+    logging.info("\tPath is {}, file is {}".format(input_path, input_name))
+
+    if "uncal" not in in_ext:
+        logging.info("\tSwitching from {} to uncal".format(in_ext))
+        uncal_name = os.path.basename(input_file).replace(in_ext, "uncal")
+        uncal_file = filesystem_path(uncal_name, check_existence=True)
+    else:
+        uncal_file = input_file
+        uncal_name = input_name
+    
+    output_file_or_files = []
+    short_name = input_name.replace("_"+in_ext, "").replace(".fits", "")
+    logging.info("\tLocking {}".format(short_name))
+    cal_lock = REDIS_CLIENT.lock(short_name)
+    have_lock = cal_lock.acquire(blocking=True)
+    if have_lock:
+        try:
+            logging.info("\t\tAcquired Lock.")
+            logging.info("\t\tCopying {} to {}".format(input_file, send_path))
+            copy_files([uncal_file], send_path)
+            if jump_pipe:
+                ramp_fit = False
+                save_fitopt = False
+                if "rate" in ext_or_exts or "rateint" in ext_or_exts:
+                    ramp_fit = True
+                if "fitopt" in ext_or_exts:
+                    save_fitopt = True
+                result = calwebb_detector1_save_jump.delay(uncal_name, ramp_fit=ramp_fit, save_fitopt=save_fitopt)
+            else:
+                result = run_calwebb_detector1.delay(uncal_name, instrument)
+            logging.info("\t\tStarting with ID {}".format(result.id))
+            processed_path = result.get()
+            logging.info("\t\tPipeline Complete")
+            if isinstance(ext_or_exts, str):
+                ext_or_exts = [ext_or_exts]
+            file_or_files = ["{}_{}.fits".format(short_name, x) for x in ext_or_exts]
+            output_file_or_files = [os.path.join(input_path, x) for x in file_or_files]
+            logging.info("\t\tCopying {} to {}".format(file_or_files, input_path))
+            copy_files([os.path.join(receive_path, x) for x in file_or_files], input_path)
+            logging.info("\t\tClearing Transfer Files")
+            to_clear = glob(os.path.join(send_path, short_name+"*")) + glob(os.path.join(receive_path, short_name+"*"))
+            for file in to_clear:
+                os.remove(file)
+        except Exception as e:
+            logging.error('\tPipeline processing failed for {}'.format(input_name))
+            logging.error('\tProcessing raised {}'.format(e))
+        finally:
+            cal_lock.release()
+            logging.info("\tReleased Lock {}".format(short_name))
+    else:
+        raise ValueError("Unable to acquire lock for {}".format(short_name))
```

Comment:
From review: lock is a `key, value` pair, should never be a case where you don't have the lock. Should never see this error raised.